docs/api.rst

```diff
@@ -1,115 +1,90 @@
 API Reference
```

Comment:
@hynek have you considered generating this document via https://pypi.org/project/sphinxcontrib-apidoc/ ?

Reply:
Why should I?

Reply:
I tend to do this in an attempt to stop maintaining an API document and just render it templated from docstrings. Specifically for "private api" — it makes many internal objects linkable too. Might be useful here with a custom template, maybe. FWIW, this could be a separate effort, I was just posting ideas to see if it'd interest you.

Reply:
hm ok I'm asking because the description is rather inconclusive as to what it gives me and it looks rather unmaintained 🤔

guess this is big enough, I'm open for separate input tho

Reply:
@hynek that project doesn't really need to be maintained actively. Basically, the Sphinx upstream has a `sphinx-apidoc` helper that generates RST documents out of the source code but needs to be run manually — you'd have to rerun it on any changes and commit those documents. `sphinxcontrib-apidoc`, OTOH, wraps that and runs this command automatically on every build allowing you to gitignore the directory with autogenerated docs and keep them in sync with the Python modules. And since the configuration mostly defines what is passed to `sphinx-apidoc` in those automatic invocations, this means that the extension does not need to be updated as often as the Sphinx upstream.

Reply:
hmmm I don't think I want such a template-based approach – if I understand it correctly.

I'm trying to editorize my API docs to have a flow of concepts and like to add examples, such that they don't have to be in docstrings which is a nightmare to do :)

Reply:
Ah, I usually like these things in docstrings. I'm curious if it's possible to combine these approaches — a top-module docstring is often useful for this.

FWIW, the flow thingy is typically separate from the reference per https://diataxis.fr/. Maybe that would be fun to discuss in the Write The Docs Slack instance :)

Reply:
I could not disagree more with this dichotomy. I'm a big fan of https://diataxis.fr/ but that doesn't mean that reference documentation should appear in random order. APIs usually build on each other and some are more important / fundamental than others. I absolutely tend to scroll through them to see what a package can do because no package under the sun has exhaustive narrative docs that cover every piece of API. Having to wade thru some weird util functions before the important stuff is positively annoying to me.

Reply:
I'm not suggesting that the order should be random, though.